plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/HiveMetastoreTableOperations.java

```diff
@@ -88,21 +88,15 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
                         .setParameter(METADATA_LOCATION_PROP, newMetadataLocation)
                         .setParameter(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation)
                         .build();
+
+                // todo privileges should not be replaced for an alter
+                PrincipalPrivileges privileges = table.getOwner().map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
+                metastore.replaceTable(database, tableName, table, privileges);
             }
             catch (RuntimeException e) {
-                try {
-                    io().deleteFile(newMetadataLocation);
```

Comment:
`deleteFile(newMetadataLocation)` should still be invoked when any of the getter instructions fails (e.g. `fromMetastoreApiTable(thriftMetastore.getTable`)
the left-over should be left behind only when the mutating call fails.

further, we should have a comment saying that we can introduce error categorization to know when such cleanup would be safe after the mutating call first. We don't need to introduce the categorization in this PR.

Reply:
I took them out based on this comment from David https://github.com/trinodb/trino/pull/10845#discussion_r815478015

Reply:
@findepi  so what should I do? add deletion here and in Glue for the case when `getter` fails ? Or be on the super safe side and not to bother with deletion here at all

Reply:
let's do what Alex did for Glue -- leave behind